modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt

```diff
@@ -862,7 +863,7 @@ sealed class IO<out E, out A> : IOOf<E, A> {
 
     companion object {
       // Internal reusable reference.
-      internal val makeUncancelable: (IOConnection) -> IOConnection = { KindConnection.uncancelable }
```

Comment:
How does cancellation work for any `F` without `KindConnection`?

Reply:
A while ago we decided not to expose `KindConnection` anymore because it's extremely error prone due to the user being responsible for `push`/`popping` cancelation tasks.

Instead we offer a couple other combinators such as `onCancel` which you can use to register a cancel token, or `cancelable`. This way we can take away that complexity from the user and we can offer a simpler API.

I've been considering merging all those APIs tho in something similar to RxJava/Flow's `create`. We can keep track of how many tasks the  user registers, and cancel them ourselves, or pop them when the effect finishes. We can also offer them a more idiomatic API (I think) compared to `cb: (Either<Throwable, A> -> Unit) ->` and `cb(Right("Hello World!"))`

```
val io = IO.create<DomainError, String> { // Receiver enabled syntax 
      success("Hello World")
      //error(DomainError)
     //exception(RuntimException("Boom"))
    onCancel(IO { println("I get printed if this task gets cancelled") })
    onCancel(IO { println("multiple tasks can be registered") })
}
```

Reply:
Anything that improves in the state of `IO.async { conn, cb ->` is welcome. Many users including myself have ran into the issue where the async constructor blindly does not return because we forget to call `cb`

Reply:
> val io = IO.create<DomainError, String> { // Receiver enabled syntax 
>      onCancel(IO { println("I get printed if this task gets cancelled") })
>      onCancel(IO { println("multiple tasks can be registered") })
> 
>       success("Hello World")
>       //error(DomainError)
>      //exception(RuntimException("Boom"))
> }

My 2 cents here regarding receiver syntax: having to guess/learn from the devs perspective which functions to call is a bit cumbersome and something I myself find more attractive from IO or Observable's creation function, where you simply get a function/interface callback parameter to call over.

Reply:
@aballano that's a very valid comment and something I was also weary about. Thanks for bringing it up ;)

So, in this case, it would be better? That'd be more in line with RxJava, and the example above is in line with `Flow`. We can also do a mix and match, so people can choose. The syntax is available implicit but also available concretely on `emitter` or whatever it's named.

```kotlin
IO.create { sink ->
   sink.success("Hello World")
  //sink.error(DomainError)
 //sink.exception(RuntimException("Boom"))
}
```